packages/compiler-cli/src/ngtsc/extended_template_diagnostics/src/template_checker.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as ts from 'typescript';
+import {TemplateTypeChecker} from '../../typecheck/api';
+import {TemplateCheck, TemplateContext} from '../api/api';
+
+const allTemplateChecks: TemplateCheck[] = [];
+
+/**
+ * Run all `TemplateChecks` for a component and return the generated `ts.Diagnostic`s.
+ * @param component the `@Component()` class from which the template is obtained
+ * @param templateTypeChecker interface to get information about template nodes
+ * @param typeChecker program's type checker
+ * @param templateChecks specific checks to be ran, default value is all the checks
+ * @returns
+ */
+export function getExtendedTemplateDiagnosticsForComponent(
+    component: ts.ClassDeclaration, templateTypeChecker: TemplateTypeChecker,
+    typeChecker: ts.TypeChecker,
+    templateChecks: TemplateCheck[] = allTemplateChecks): ts.Diagnostic[]|undefined {
```

Comment:
```suggestion
    templateChecks: TemplateCheck[] = allTemplateChecks): ts.Diagnostic[]|undefined {
```

Same question about `null` vs `undefined`
// cc @alxhub

Reply:
@dgp1130 suggested to limit the scope to just `ts.Diagnostic[]` and return an empty array in cases where no diagnostics were found.